packages/rest/src/router/redirect-route.ts

```diff
@@ -39,4 +46,24 @@ export class RedirectRoute implements RouteEntry, ResolvedRoute {
       this.targetLocation
     }"`;
   }
+
+  private _getHost(request: Request) {
+      return request.get('x-forwarded-host') || request.headers.host;
+  }
+
+  private _getProtocolForRequest(request: Request) {
+      return (
+          (request.get('x-forwarded-proto') || '').split(',')[0] ||
+          request.protocol ||
+          'http'
+      );
+  }
+
+   private _getBasePath(request: Request) {
+        let basePath = '';
+        if (request.baseUrl && request.baseUrl !== '/') {
```

Comment:
`request.baseUrl` is set by Express when a middleware is mounted via `app.use(basePath, handler)`. It's a different concept from LB4's `basePath` option.

Eventually, we need to support both. Let's work in small increments please and leave `baseUrl` out of scope of this pull request.